WoWsShipBuilder.Web/Pages/ShipComparison.razor

```diff
@@ -715,34 +720,163 @@
 
     private void OpenShipStats()
     {
-        SetupAndOpenNewTab("ship?shipIndexes=" + string.Join(",", ViewModel!.SelectedShipList.Select(x => x.ShipIndex)));
+        SetupAndOpenNewTab("ship?shipIndexes=" + string.Join(",", ViewModel!.SelectedShipList.Select(x => x.Value.Ship.Index)));
     }
 
     private void OpenBallisticCharts()
     {
-        SetupAndOpenNewTab("charts?shipIndex=" + string.Join(",", ViewModel!.SelectedShipList.Select(x => x.ShipIndex)));
+        SetupAndOpenNewTab("charts?shipIndex=" + string.Join(",", ViewModel!.SelectedShipList.Select(x => x.Value.Ship.Index)));
     }
 
     private void OpenAccelerationCharts()
     {
-        SetupAndOpenNewTab("acceleration-charts?shipIndexes=" + string.Join(",", ViewModel!.SelectedShipList.Select(x => x.ShipIndex)));
+        SetupAndOpenNewTab("acceleration-charts?shipIndexes=" + string.Join(",", ViewModel!.SelectedShipList.Select(x => x.Value.Ship.Index)));
     }
 
-    private async Task UpdateUrl()
+    private void ViewShipStats(ShipBuildContainer container)
+    {
+        SessionStateCache.SetBuildTransferContainers(container);
+        NavManager.NavigateTo("ship?shipIndexes=" + container.Ship.Index);
+    }
+
+    private bool IsColumnHidden(string section, string columnTitle)
     {
-        QueryBuilder url = new()
+        return !ViewModel!.SelectedDataSection.ShipComparisonDataSectionToString().Equals(section) || hiddenColumns.TryGetValue($"{section}_{columnTitle}", out bool hidden) && hidden;
+    }
+
+    private static char ClassToUrlIndex(ShipClass shipClass)
```

Comment:
Consider extracting these mappings to a separate utility class so that all index/letter mappings are in one place. GameDataHelper in Core might be a good fit